pkg/kubelet/kubelet.go

```diff
@@ -2419,3 +2430,20 @@ func isSyncPodWorthy(event *pleg.PodLifecycleEvent) bool {
 	// ContainerRemoved doesn't affect pod state
 	return event.Type != pleg.ContainerRemoved
 }
+
+func redirectChecker(followNonLocalRedirects bool) func(*http.Request, []*http.Request) error {
+	if followNonLocalRedirects {
+		return nil // Use the default http client checker.
+	}
+
+	return func(req *http.Request, via []*http.Request) error {
+		if req.URL.Hostname() != via[0].URL.Hostname() {
+			return http.ErrUseLastResponse
+		}
+		// Default behavior: stop after 10 redirects.
+		if len(via) >= 10 {
+			return errors.New("stopped after 10 redirects")
+		}
+		return nil
+	}
+}
```

Comment:
nit: It looks like this is never called with `followNonLocalRedirects = true`? In that case, you could just do something like this instead of using a closure:
```suggestion
func requireSameHostRedirects(*http.Request, []*http.Request) error {
	if req.URL.Hostname() != via[0].URL.Hostname() {
		return http.ErrUseLastResponse
	}
	// Default behavior: stop after 10 redirects.
	if len(via) >= 10 {
		return errors.New("stopped after 10 redirects")
	}
	return nil
}
```